### PR TITLE
Make things compile correctly

### PR DESCRIPTION
**Pull request by jcmuller**
You can merge this PR into your WIP branch if you like these changes.

- Cleaned up a lot of linter errors.
- Tests pass (didn't confirm that the coverage is adequate, though)
- Added TODOs with my questions.
- Aligned some structs.

### CODE REVIEW

File: neoq.go [hunk 1]
```diff
@@ -22,7 +14,7 @@ import (
 // Neoq interface is Neoq's primary API
 type Neoq interface {
 	// Enqueue queues jobs to be executed asynchronously
-	Enqueue(ctx context.Context, job jobs.Job) (jobID int64, err error)
+	Enqueue(ctx context.Context, job *jobs.Job) (jobID int64, err error)
```
Comment (jcmuller) [inline neoq.go:17]: `job` is a pretty big object. Pass it by reference.

File: handler/handler.go [hunk 2]
```diff
@@ -70,33 +76,32 @@ func NewHandler(f HandlerFunc, opts ...HandlerOption) (h Handler) {
...
+		Deadline(DefaultHandlerDeadline)(&h)
 	}
 
 	return
 }
 
-// WithHandlerContext creates a new context with the job and transaction set
-func WithHandlerContext(ctx context.Context, v nctx.HandlerCtxVars) context.Context {
+// WithContext creates a new context with the job and transaction set
+func WithContext(ctx context.Context, v nctx.HandlerCtxVars) context.Context {
```
Comment (jcmuller) [inline handler/handler.go:91]: Sorry, I just went ahead and removed the stutter from this package

File: neoq.go (hunk 1 above)
Comment (acaloiaro) [reply neoq.go:17]: Yeah, and with arbitrary payload sizes it would mean a lot of copying.

**Review by acaloiaro — APPROVED** [review]
:boat:

**Pull request merged**